### DISCUSSION TITLE
Initializing dynamic memory based off of a parameter

### DISCUSSION THREAD

**Discussion by zimri-leisher** [nasa/fprime#2721] [Q&A] [answered]
I have a component in which I want to create a heap array of a size defined in a parameter. Because heap memory allocation should be kept to initialization, I want this to happen as soon as possible after parameters are loaded from the file.
Any advice on how I should implement this? At the moment, I'm just going in my `setupTopology` function, and after `loadParameters` are called I'm manually calling an `init` function on the component. Is there a fancier way to do this that works more cleanly with FPrime?

**Comment by timcanham** [comment, marked as answer]
I would do it the exact same way for what it's worth. After `loadParameters()` but before the software starts cycling.

**Reply by zimri-leisher** [reply]
Good enough for me then. Just wanted to make sure I wasn't missing a more obvious solution
